### ISSUE TITLE
provide official guidelines for the use of "docker" & logos

### ISSUE THREAD

**Issue by unclejack** [moby/moby#15974]
Many FOSS projects have docker in their name. Someone who wishes to add support for docker to their project sometimes creates a docker-foo repository (let's call the project they've added Docker support to "foo").
This leads to a few issues:
- searching for projects which use Docker is very difficult as there are thousands of projects with docker in their name
- multiple projects with very similar names are created, sometimes they differ by a single letter or through pluralization (docker-foo vs dockers-foo, docker-foo vs docker-foos)
- they make themselves a potential target for a cease & desist letter (!)

Another issue is the use of the Docker logo which makes third party projects, products and websites appear to be official ones (!). Even if we ignore any potential request for the logo to be taken down as a potential problem for those using it, there's still the problem of making things confusing for users who happen to come across a site which uses the logo and makes people think that site is official.

The same issues apply to commercial products. Should there be a special "partner" application process to obtain permission to do things differently, that process also requires the same kind of thorough documentation.

These issues have to be addressed by creating a set of guidelines in plain English without any legalese so that anyone who wants to create an open source or closed source project (or product) can do so without requiring legal counsel.

Since this is an open source project, the FOSS community assumes they can use the name freely for open source projects and they should know in advance if there's anything specific to this project which is different.

Checklist for items to be documented:
- when is it OK to use the docker name in a project or product's name
- how are you allowed to use the docker name
- provide examples of what is OK and NOT OK to do
- explain how and when the logo can be used
- explain how to request special permission for the logo to be used another way
- explain how to request special permission to make "docker" part of the name of a product/project
- provide guidelines for commercial products
- provide official contact for other questions
- should this be desirable, provide an alternate "ecosystem" logo to be used freely

**Comment by estesp** [comment]
I assume this page covers at least some of the open questions: https://www.docker.com/brand-guidelines

**Comment by unclejack** [comment]
@estesp That's very good. I didn't know it even existed. It's unfortunate that it's not easy to come across that unless you're looking for such a page.

Similar guidelines should be checked into the project's repository. I'm not going to assume everyone who drops by https://github.com/docker/docker also visits that page or that those are exhaustive guidelines regarding best practices in Docker related FOSS projects.

**Comment by rdsubhas** [comment]
@estesp @unclejack The page is good, but there are lots of practical problems.

The brand guidelines says **Do not** Use Docker in the name of your website, application or product. Well, if I'm publishing an image for Docker, its really difficult to name it without having "docker" somewhere, say "docker-alpine-node" or "alpine-docker-node" or something like that.

I just got a friendly email from one of your community managers as well regarding my `docker-menubar` side project, asking it to be renamed `menubar-for-docker`. I can't see how renaming `docker-menubar` to `menubar-for-docker` would make it compliant with the branding guidelines page, which says don't use "docker" anywhere in the name.

This is a huge point of confusion for image publishers, addon creators and simple, small OSS projects like myself. Some more clarity on this would be really needed.

**Comment by LK4D4** [comment]
@thaJeztah I think this is an important issue. I wonder if you could take care of it.

**Comment by thaJeztah** [comment]
I asked the trademark team for some clarification earlier today to make it easier to understand what's "ok" and what's "not ok" to do. I'll post an update if more information is available

w.r.t.

>  I can't see how renaming docker-menubar to menubar-for-docker would make it compliant with the branding guidelines page

it's correct that _both_ are (probably) not compliant with the guidelines, but (from some internal discussions), a project / product named `docker-something` is more likely to be confused for an "official" docker product (because it follows the same naming as `docker-compose`, `docker-machine`, etc.). However, it's understandable that a project may want to refer to what something is used for ("docker"), so where possible, the team tries to be "flexible".

(note that I'm not a lawyer, and not involved in the trademark/legal team, so the above is my interpretation)

**Comment by rdsubhas** [comment]
@thaJeztah thanks for the update :) I checked the page again, and it now reads:

> Use the word Docker as **the first word** in the name of your website or ...

which is helpful. Thanks!

**Comment by thaJeztah** [comment]
I got an initial reply from Docker Inc., I'll link it here: https://github.com/docker/docker/issues/26556#issuecomment-247294749

**Comment by thaJeztah** [comment]
Closing this one, with https://www.docker.com/brand-guidelines being available on the website, and https://github.com/moby/moby/issues/26556#issuecomment-247294749 having additional information.
